src/unix/syscall.c

```diff
@@ -2605,6 +2637,16 @@ void init_syscalls(tuple root)
     syscall = syscall_handler;
     syscall_io_complete = closure(h, syscall_io_complete_cfn);
     io_completion_ignore = closure(h, io_complete_ignore);
+    vector hostname_v = split(h, alloca_wrap_cstring("etc/hostname"), '/');
+    tuple hostname_t = resolve_path(root, hostname_v);
+    buffer b;
```

Comment:
This vector-of-buffers dealloc seems generally useful - maybe it could accompany split(), join(), etc. in vector.h?

Reply:
OK, I wrote a `split_dealloc()` function in vector.h that deallocates what is allocated by split().